src/components/UserInfo/UserInfoCard/index.tsx

```diff
@@ -0,0 +1,228 @@
+        return b.points - a.points;
+      });
+    const rank = sortedUserList.findIndex((data) => data.id === id) + 1;
+    return rank;
+  }, [id]);
+
+  // TODO: 유저가 작성한 퀴즈 정보 비동기 API요청 필요
+  const getQuiz = () => {
+    const modifiedQuiz = userQuizList.map((quiz) => {
+      const customData = JSON.parse(quiz.title);
+      return {
+        id: quiz._id,
+        category: customData.tag,
+      };
+    });
+    return modifiedQuiz;
+  };
+
+  const level = Math.floor(userData.totalExp / MAXEXP) + 1;
+  const currentExp = userData.totalExp - (level - 1) * MAXEXP;
+  const expPercent = Math.floor((currentExp / MAXEXP) * 100);
+
+  const getImage = useCallback(() => {
+    let selectedId = Breakpoints.imageBreakpoints[0].imageId;
+    Breakpoints.imageBreakpoints.forEach((breakpoint) => {
+      if (level >= breakpoint.level) {
+        selectedId = breakpoint.imageId;
+      }
+    });
+    return selectedId;
+  }, [level]);
+
+  interface badgeType {
+    id: string;
+    color?: string;
+    content: string;
+  }
+
+  const getBadges = () => {
+    const badges: badgeType[] = [];
+
+    // 레벨 별 뱃지
+    let levelBadgeId = 0;
+    Breakpoints.levelBreakpoints.forEach((badge, index) => {
+      if (level > badge.level) {
+        levelBadgeId = index;
+      }
+    });
+
+    badges.push({
+      id: `badgeLevel${levelBadgeId}`,
+      color: Breakpoints.levelBreakpoints[levelBadgeId].color,
+      content: `${
+        levelBadgeId !== 0
+          ? `내가 레벨 ${Breakpoints.levelBreakpoints[levelBadgeId].level} 이라니!`
+          : `뉴비`
+      }`,
+    });
+
+    // 출제한 퀴즈 카테고리 별 뱃지
+    const categoryMap = new Map();
+
+    getQuiz().forEach((quiz) => {
+      const { category } = quiz;
+      const count = categoryMap.get(category);
+      if (count) {
+        categoryMap.set(category, count + 1);
+      } else {
+        categoryMap.set(category, 1);
+      }
+    });
+
+    const categoryQuiz = Array.from(categoryMap).sort((a, b) => {
+      if (a[1] <= b[1]) return 1;
+      return -1;
+    });
+
+    categoryQuiz.forEach((quiz, index) => {
+      if (quiz[1] >= 50) {
+        badges.push({
+          id: `badgeCategory${index}`,
+          content: `${quiz[0]}을 정복한 자`,
+        });
+      } else if (quiz[1] >= 10) {
+        badges.push({
+          id: `badgeCategory${index}`,
+          content: `${quiz[0]}을 시작한 자`,
+        });
+      }
+    });
+
+    // 댓글과 좋아요로 뱃지 가져오기
+    Breakpoints.commentBreakpoints.forEach((breakpoint, index) => {
+      if (breakpoint.exact) {
+        if (userData.comments.length === breakpoint.count) {
+          badges.push({
+            id: `badgeComment${index}`,
+            color: breakpoint.color,
+            content: breakpoint.text,
+          });
+        }
+      } else if (userData.comments.length >= breakpoint.count) {
+        badges.push({
+          id: `badgeComment${index}`,
+          color: breakpoint.color,
+          content: breakpoint.text,
+        });
+      }
+    });
```

Comment:
뱃지추가가 반복되기도 하고, 
이렇게 하면, 뱃지는 무엇이고, 추가로직은 무엇이다, 구분할 수 있어서 가독성이 더 좋다고 생각합니다.
```suggestion
  Breakpoints.commentBreakpoints.forEach((breakpoint, index) => {
      const badge = {
        id: `badgeComment${index}`,
        color: breakpoint.color,
        content: breakpoint.text,
      };
      if (breakpoint.exact) {
        if (userData.comments.length === breakpoint.count) {
          badges.push(badge);
        }
      } else if (userData.comments.length >= breakpoint.count) {
        badges.push(badge);
      }
    });
```

Reply:
if문 중복의 경우, 벗겨보려고 시도했는데, 아래 로직들에 공통적으로 적용하기도 까다로울 것 같아서, 가장 비즈니스 로직에 맞는 지금 방식도 괜찮다고 느껴지네요 ^^. 따라 읽어 내려가면 되니까요.

Reply:
이 방식이 좀 더 한눈에 들어오네요!! 이 부분 구현할 때 중복이 마음에 걸렸었는데 badge를 먼저 이것이다! 이렇게 정의하니까 해결되네요... 감사합니다!!  
if 중복 같은 경우.. 추후 리팩토링 시 고민해볼게요!